unix/w0vncserver/PipeWirePixelBuffer.cxx

```diff
@@ -106,20 +111,56 @@ void PipeWirePixelBuffer::processFrame(spa_buffer* buffer)
     return;
   }
 
+  // Clamp values as they can be outside of the buffer
+  x = std::max(0, accumulatedDamage.get_bounding_rect().tl.x);
+  y = std::max(0, accumulatedDamage.get_bounding_rect().tl.y);
+  w = accumulatedDamage.get_bounding_rect().width();
+  h = accumulatedDamage.get_bounding_rect().height();
+
+  if (x + w > width())
+    w = width() - x;
+  if (y + h > height())
+    h = height() - y;
```

Comment:
We have intersect operation for that. This precise issue is handled in other parts of the code.